src/jdk.incubator.vector/linux/native/libsleef/lib/vector_math_rvv.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2024, Rivos Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+// On riscv, sleef vector apis depend on native vector intrinsic, which is supported on
+// some compiler, e.g. gcc 14+.
+// __riscv_v_intrinsic is used to tell if the compiler supports vector intrinsic.
+//
+// At compile-time, if the current compiler does support vector intrinsics, bridge
+// functions will be built in the library. In case the current compiler doesn't support
+// vector intrinsics (gcc < 14), then the bridge functions won't be compiled.
+// At run-time, if the library is found and the bridge functions are available in the
+// library, then the java vector API will call into the bridge functions and sleef.
+
+#ifdef __riscv_v_intrinsic
+
+#include <stdint.h>
+
+#include <riscv_vector.h>
+
+#include "../generated/misc.h"
+#include "../generated/sleefinline_rvvm1.h"
+
+#include <jni.h>
+
+// We maintain an invariant in java world that default dynamic rounding mode is RNE,
+// please check JDK-8330094, JDK-8330266 for more details.
+// Currently, sleef source on riscv does not change rounding mode to others except
+// of RNE. But we still think it's safer to make sure that after calling into sleef
+// the dynamic rounding mode is always RNE.
+
+#ifdef DEBUG
```

Comment:
Question: Should we check for `NDEBUG` macro here instead? I see checks for this macro in the original SLEEF code.

```
#ifndef NDEBUG
#define CHECK_FRM   __asm__ __volatile__ (     \
    "    frrm   t0              \n\t"          \
    "    beqz   t0, 2f          \n\t"          \
    "    csrrw  x0, cycle, x0   \n\t"          \
    "2:                         \n\t"          \
    : : : "memory" );
#else
#define CHECK_FRM
#endif
```

Reply:
`NDEBUG` is only used in sleefdp.c which is the original sleef code, and we don't use that file in jdk directly, in java.base module of jdk it uses `DEBUG` but not use `NDEBUG`.
Based on above information, I think `DEBUG` is better.

Reply:
Sounds good. I suppose this `DEBUG` macro will only be defined for debug builds, right?

Reply:
Yes, I tested on both release and fastdebug version.